### DISCUSSION TITLE
Syntax error on valid query, should be querytime error

### DISCUSSION THREAD

**Discussion by quinnlangille** [supabase/supabase#16943] [Contribute to Supabase]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

When trying to create a unique index on a table with data that would violate the constraint, supabase reports a query syntax error vs a runtime error. The following query:

```postgres
ALTER TABLE address 
ADD CONSTRAINT address_info_combined_idx 
UNIQUE NULLS NOT DISTINCT (street_number, street_name, unit_number, city, state, country, postal_code)
```

Returns `SYNTAX ERROR AT OR NEAR NULLS`, but when run in a 3rd party tool against the same DB I get "Could not create unique index address_info_combined_idx" (which admittedly is also not a great error, but less of a red herring on a query with valid syntax

## To Reproduce

Steps to reproduce the behaviour, please provide code snippets or a repository:

1. Add two rows with the same value in a given column
2. Attempt to add a unique constraint on that column in supabase SQL editor

## Expected behaviour

A runtime error vs a syntax error

## System information

- OS: macOS
- Browser (if applies) Chrome

## Additional context

Running PG 15.3

**Comment by GaryAustin1** [comment]
This is because the SQL editor still does not have handling for new Postgres 15 syntax.  
Duplicate of:
https://github.com/supabase/supabase/issues/13267 (Valid query containing UNIQUE NULLS NOT DISTINCT is not validated)
https://github.com/supabase/supabase/issues/13460
https://github.com/supabase/supabase/issues/11706

**Comment by AshwanthramKL** [comment]
Hey, can I work on this?
I believe in removing the "NULLS NOT DISTINCT" part and adding a WHERE clause to check for null values in the columns involved in the unique constraint, it will only apply to non-null values. 

As a result, Supabase would report a runtime error instead of a syntax error when attempting to add a unique constraint with data that violates the constraint.
